lib/CompilerImpl/ClangLibCompiler.cpp

```diff
@@ -235,6 +235,8 @@ ClangLibCompiler::runOptimizer(const std::filesystem::path &src_IR,
     argv.push_back(arg.c_str());
     log_str = log_str + arg + " ";
   }
+  argv.push_back(src_IR.c_str());
```

Comment:
The previous work used to tamper quite a lot the LLVM opt files. 
99% of the tampering was to avoid to supply the file name.
That made me lose quite a lot of hours to figure out which changes had been done to the LLVM opt files and why they had been made (because many functionalities were pruned for no apparent reason that it would have required to refactor heavily the code).
As I realized that all of the useful changes were made just to avoid providing the filename to be used (it took me some quite a long list of diffs to be inspected) , I opted for an easier approach: include LLVM opt files almost untouched, with commented edits, to keep maintainability on further updates. So that's why we now supply the IR source file.